pkg/virt-handler/device-manager/usb_device_test.go

```diff
@@ -10,212 +10,225 @@ import (
+				[]v1.USBHostDevice{
+					{
+						ResourceName: resourceName1,
+						Selectors: []v1.USBSelector{
+							{
+								Vendor:  fmt.Sprintf("%x", usbs[0].Vendor),
+								Product: fmt.Sprintf("%x", usbs[0].Product),
+							},
 						},
 					},
-				},
-				{
-					ResourceName:             resourceName2,
-					ExternalResourceProvider: true,
-					Selectors: []v1.USBSelector{
-						{
-							Vendor:  fmt.Sprintf("%x", usbs[1].Vendor),
-							Product: fmt.Sprintf("%x", usbs[1].Product),
+					{
+						ResourceName:             resourceName2,
+						ExternalResourceProvider: true,
+						Selectors: []v1.USBSelector{
+							{
+								Vendor:  fmt.Sprintf("%x", usbs[1].Vendor),
+								Product: fmt.Sprintf("%x", usbs[1].Product),
+							},
 						},
 					},
 				},
-			},
-			map[string][]*PluginDevices{
-				resourceName1: []*PluginDevices{
-					newPluginDevices(resourceName1, 0, []*USBDevice{usbs[0]}),
-				},
-			},
-		),
-		Entry("Should ignore a config with same USB selector",
-			[]v1.USBHostDevice{
-				{
-					ResourceName: resourceName1,
-					Selectors: []v1.USBSelector{
-						{
-							Vendor:  fmt.Sprintf("%x", usbs[0].Vendor),
-							Product: fmt.Sprintf("%x", usbs[0].Product),
-						},
+				map[string][]*PluginDevices{
+					resourceName1: []*PluginDevices{
+						newPluginDevices(resourceName1, 0, []*USBDevice{usbs[0]}),
 					},
 				},
-				{
-					ResourceName: resourceName2,
-					Selectors: []v1.USBSelector{
-						{
-							Vendor:  fmt.Sprintf("%x", usbs[0].Vendor),
-							Product: fmt.Sprintf("%x", usbs[0].Product),
+			),
+			Entry("Should ignore a config with same USB selector",
+				[]v1.USBHostDevice{
+					{
+						ResourceName: resourceName1,
+						Selectors: []v1.USBSelector{
+							{
+								Vendor:  fmt.Sprintf("%x", usbs[0].Vendor),
+								Product: fmt.Sprintf("%x", usbs[0].Product),
+							},
 						},
-						{
-							Vendor:  fmt.Sprintf("%x", usbs[1].Vendor),
-							Product: fmt.Sprintf("%x", usbs[1].Product),
+					},
+					{
+						ResourceName: resourceName2,
+						Selectors: []v1.USBSelector{
+							{
+								Vendor:  fmt.Sprintf("%x", usbs[0].Vendor),
+								Product: fmt.Sprintf("%x", usbs[0].Product),
+							},
+							{
+								Vendor:  fmt.Sprintf("%x", usbs[1].Vendor),
+								Product: fmt.Sprintf("%x", usbs[1].Product),
+							},
 						},
 					},
 				},
-			},
-			map[string][]*PluginDevices{
-				resourceName1: []*PluginDevices{
-					newPluginDevices(resourceName1, 0, []*USBDevice{usbs[0]}),
+				map[string][]*PluginDevices{
+					resourceName1: []*PluginDevices{
+						newPluginDevices(resourceName1, 0, []*USBDevice{usbs[0]}),
+					},
 				},
-			},
-		),
-	)
+			),
+		)
+	})
+	Context("Host has no USB Devices", func() {
+		It("Should not panic", func() {
+			oldPathToUSBDevices := PathToUSBDevices
+			PathToUSBDevices = "/this/path/does/not/exist"
+			DeferCleanup(func() {
+				PathToUSBDevices = oldPathToUSBDevices
```

Comment:
Please initialize this variable in a BeforeEach` section for the tests

Reply:
I have moved the code to a `BeforeEach` Section.

Reply:
I mean without the `DeferCleanup`. You can add another `BeforeEach` in the other context with the valid path.

Reply:
In that case i would rather just drop the `DeferCleanup` code entirely, since the other context does not call the `discoverPluggedUSBDevices` function at all, instead overriding it with their own implementation to ensure consistent test results.
I mainly added `DeferCleanup` to not leave the unit in a weird state afterwards, so setting the variable to it's default value should not need to happen in a `BeforeEach`

Reply:
Sounds good